### PR TITLE
Add MessageAwaiter contrib

### PR DESCRIPTION
**Pull request by ssine**

### CODE REVIEW

**Comment by huan** [comment]
It's great to see your contribution for adding a new plugin!

Could you please file an issue to description this plugin in details about what problem is it trying to solve, and why it is a good solution, so that we can review this new feature , refine the design, and finally decide whether to accept it.

Looking forward to learn from your post, thanks!

**Comment by ssine** [comment]
Thanks for the reply! the [issue](https://github.com/wechaty/wechaty-plugin-contrib/issues/13) is opened, hope we can have more discussion there.

File: src/contrib/message-awaiter.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    log.verbose('WechatyPluginContrib', 'MessageAwaiter installing on %s ...', wechaty)
+
+    wechaty.waitForMessage = async (args: MessageAwaiterArgs): Promise<Message> => {
+      let { contactId, roomId, regex, timeoutSecond } = args
+      let waitTime = new Date()
+
+      return new Promise<Message>((resolve, reject) => {
+
+        let callback = async (message: Message) => {
+          let messageFrom = message.from()
```
Comment (huan) [inline src/contrib/message-awaiter.ts:40]: Then we can use the matchers helper functions like this:

```ts
if (!matchContact(message.from)) { return }
if (!matchRoom(message.room()) { return }
if (!matchString(message.text()) { return }
```

**Comment by ssine** [comment]
I have switched to use mathcers for message pattern matching, thanks for your detailed directions!

**Comment by huan** [comment]
Your updates looks good!

Please resolve the conflicting in the PR before we can merge it, thank you very much!

**Comment by ssine** [comment]
I've merged with newest master brance and fixed the naming, looks good now

**Comment by huan** [comment]
Thank you very much for your contribution!

I have invited you to join the Wechaty Contributors team on GitHub, please accept it by following the below message:

> You've invited sine to Wechaty! They'll be receiving an email shortly. They can also visit https://github.com/wechaty to accept the invitation.

And we have a Wechaty Contributors Group on WeChat, please contact @lijiarui to join us. 

Cheers!

**Comment by ssine** [comment]
OK, glad to be one of the contributors, cheers!

**Comment by lijiarui** [comment]
Congratulations! 🎉 🎉 🎉

You become a wechaty contributor successfully! 👏 👏 👏

Please:

- [ ]  Add my wechat: ruirui_0914 and send me this pr link. I will invite you into Wechaty Contributor Room
- [ ] Update your profile here: [Contributor List](https://github.com/wechaty/wechaty/wiki/Contributors), then other contributors will know you better!
- [ ]  Join wechaty org by clicking https://github.com/wechaty

**Pull request merged**